staging/src/k8s.io/apiserver/pkg/cel/cidr.go

```diff
@@ -33,7 +33,7 @@ type CIDR struct {
 }
 
 var (
-	CIDRType = cel.OpaqueType("net.CIDR")
+	CIDRType = cel.ObjectType("net.CIDR")
```

Comment:
general question: are there any typechecking implications of this change (here and elsewhere) that would impact existing expressions?

general question: where we have the top-level functions / param / return types that we declare, can we accumulate / iterate over the types in a unit test that would notice if no cost estimates for basic things like equality are defined?

Reply:
> general question: are there any typechecking implications of this change (here and elsewhere) that would impact existing expressions? is there a way to prevent accidental use of opaquetype in the future if that interferes with cost estimation?

I've backed out this switch from Opaque->Object. There was an annoying limitation in  OpaqueType not matching in switch statements and I worked around it by switching types, but forgot to clean it up. It would be tricky to notice the difference in CEL (I can't think of a way to do it), but I don't think it would be impossible, so keeping opaque here is the way to go.

> general question: where we have the top-level functions / param / return types that we declare, can we accumulate / iterate over the types in a unit test that would notice if no cost estimates for basic things like equality are defined?

I dug into this. I think it can be done but it's going to be a significantly larger change.  OK if I split this out?  I've put a placeholder list in for now to show that we can detect these in cost_test.go.  https://github.com/kubernetes/kubernetes/issues/126365